File: src/java/magmac/app/lang/node/JavaConstructor.java
package magmac.app.lang.node;

import magmac.api.Option;
import magmac.app.compile.error.CompileResult;
import magmac.app.compile.node.Node;
import magmac.app.lang.Destructors;

public record JavaConstructor(String name) implements JavaMethodHeader {
    public static Option<CompileResult<JavaMethodHeader>> deserialize(Node node) {
        return Destructors.destructWithType("constructor", node).map(constructor -> constructor.withString("name").complete(JavaConstructor::new));
    }
}

File: src/web/magmac/app/lang/java/JavaConstructor.ts
import { Option } from "../../../../magmac/api/Option";
import { CompileResult } from "../../../../magmac/app/compile/error/CompileResult";
import { Node } from "../../../../magmac/app/compile/node/Node";
import { Destructors } from "../../../../magmac/app/lang/Destructors";
export class JavaConstructor {
	deserialize(node : Node) : Option<CompileResult<JavaMethodHeader>> {return Destructors.destructWithType( "constructor", node).map( 0);;}
}
